prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+from typing import Any
+
+from utils.request import HTTPMethod, make_http_request
+
+from .constants import VariableConstants, VariableType
+
+logger = logging.getLogger(__name__)
+
+
+class VariableService:
+
+    @staticmethod
+    def replace_static_variable(
+        prompt: str, structured_output: dict[str, Any], variable: str
+    ) -> str:
+        output_value = VariableService.check_static_variable_run_status(
+            structure_output=structured_output, variable=variable
+        )
+        static_variable_marker_string = "".join(["{{", variable, "}}"])
+        if not output_value:
+            return prompt
+        replaced_prompt: str = VariableService.replace_generic_string_value(
+            prompt=prompt, variable=static_variable_marker_string, value=output_value
+        )
+
+        return replaced_prompt
+
+    @staticmethod
+    def check_static_variable_run_status(
+        structure_output: dict[str, Any], variable: str
+    ) -> Any:
+        output = None
+        try:
+            output = structure_output[variable]
+            return output
+        except KeyError:
+            logger.warn(
+                f"Prompt with {variable} is not executed yet."
+                " Unable to replace the variable"
+            )
+            return output
+
+    @staticmethod
+    def replace_generic_string_value(prompt: str, variable: str, value: str) -> str:
+        replaced_prompt = re.sub(variable, value, prompt)
+        return replaced_prompt
+
+    @staticmethod
+    def identify_and_fetch_variable_type(variable: str) -> VariableType:
```

Comment:
```suggestion
    def identify_variable_type(variable: str) -> VariableType:
```
NIT: You can make the function less verbose without compromising on information transferred, besides there's no real `fetch` happening